backend/src/services/billing/billing.service.ts

```diff
@@ -0,0 +1,201 @@
+import StripeApi from 'stripe';
+
+import {
+  ExceptionMessage,
+  PaymentUnit,
+  TransactionStatus,
+} from '~/common/enums/enums';
+import {
+  BillingInitHoldStudentPaymentArgumentsDto,
+  BillingReplenishArgumentsDto,
+  StripeReplenishArgumentsDto,
+  TransactionCreateArgumentsDto,
+  TransactionGetAllItemResponseDto,
+} from '~/common/types/types';
+import { BillingError } from '~/exceptions/exceptions';
+import {
+  transaction as transactionServ,
+  user as userServ,
+  userDetails as userDetailsServ,
+} from '~/services/services';
+
+type Constructor = {
+  secretKey: string;
+  apiVersion: string;
+  transactionService: typeof transactionServ;
+  userService: typeof userServ;
+  userDetailsService: typeof userDetailsServ;
+};
+
+class Billing {
+  #transactionService: typeof transactionServ;
+
+  #userService: typeof userServ;
+
+  #userDetailsService: typeof userDetailsServ;
+
+  #stripe: StripeApi;
+
+  private static BILLING_CURRENCY = 'usd';
+
+  private static DEFAULT_STUDYING_PRICE_COEFFICIENT = 0.5;
+
+  private static NEW_USER_BALANCE_AFTER_WITHDRAW = 0;
+
+  public constructor({
+    transactionService,
+    userService,
+    userDetailsService,
+    secretKey,
+    apiVersion,
+  }: Constructor) {
+    this.#stripe = new StripeApi(secretKey, {
+      apiVersion: apiVersion as '2022-08-01',
```

Comment:
Why don't you declare `apiVersion` as `2022-08-01` on Constructor type? 🤔 
```suggestion
      apiVersion
```